packages/arb-token-bridge-ui/tests/e2e/specs/withdrawCctp.cy.ts

```diff
@@ -4,179 +4,237 @@
 
 import { CommonAddress } from 'packages/arb-token-bridge-ui/src/util/CommonAddressUtils'
 import { formatAmount } from '../../../src/util/NumberUtils'
-import {
-  getInitialERC20Balance,
-  getL1TestnetNetworkConfig,
-  getL2TestnetNetworkConfig
-} from '../../support/common'
+import { shortenAddress } from '../../../src/util/CommonUtils'
 
 describe('Withdraw USDC through CCTP', () => {
-  const USDCAmountToSend = Number((Math.random() * 0.001).toFixed(6)) // randomize the amount to be sure that previous transactions are not checked in e2e
-
   // Happy Path
   context('User is on L2 and imports USDC', () => {
-    let l1USDCBal: string
-    let l2USDCBal: string
+    let USDCAmountToSend: number
 
     // log in to metamask before withdrawal
     beforeEach(() => {
       cy.fundUserWalletEth('L2')
       cy.fundUserUsdcTestnet('L2')
       cy.resetCctpAllowance('L2')
-
-      const address = Cypress.env('ADDRESS')
-      getInitialERC20Balance({
-        tokenAddress: CommonAddress.Sepolia.USDC,
-        multiCallerAddress: getL1TestnetNetworkConfig().multiCall,
-        rpcURL: Cypress.env('ETH_SEPOLIA_RPC_URL'),
-        address
-      }).then(
-        val =>
-          (l1USDCBal = formatAmount(val, {
-            symbol: 'USDC',
-            decimals: 6
-          }))
-      )
-      getInitialERC20Balance({
-        tokenAddress: CommonAddress.ArbitrumSepolia.USDC,
-        multiCallerAddress: getL2TestnetNetworkConfig().multiCall,
-        rpcURL: Cypress.env('ARB_SEPOLIA_RPC_URL'),
-        address
-      }).then(
-        val =>
-          (l2USDCBal = formatAmount(val, {
-            symbol: 'USDC',
-            decimals: 6
-          }))
-      )
+      USDCAmountToSend = Number((Math.random() * 0.001).toFixed(6)) // randomize the amount to be sure that previous transactions are not checked in e2e
+
+      cy.login({ networkType: 'L2', networkName: 'arbitrum-sepolia' })
+      context('should show L1 and L2 chains, and ETH correctly', () => {
+        cy.findByRole('button', { name: /From: Arbitrum Sepolia/i }).should(
+          'be.visible'
+        )
+        cy.findByRole('button', { name: /To: Sepolia/i }).should('be.visible')
```

Comment:
I'd reuse the entire thing if possible also, to reduce the cognitive load here. It would be much easier to read `searchAndSelectToken(address)` and have all this code abstracted.

Reply:
Good suggestion, updated.